tests/api.rs

```diff
@@ -1744,50 +1758,19 @@ fn quic_handshake() {
     let server_hs = step(&mut server, &mut client).unwrap().unwrap();
     assert!(server.get_early_secret().is_none());
     let client_hs = step(&mut client, &mut server).unwrap().unwrap();
-    assert_eq!(server_hs, client_hs);
+    assert!(equal_secrets(&server_hs, &client_hs));
     assert!(client.is_handshaking());
     let server_1rtt = step(&mut server, &mut client).unwrap().unwrap();
     assert!(!client.is_handshaking());
     assert_eq!(client.get_quic_transport_parameters(), Some(server_params));
     assert!(server.is_handshaking());
     let client_1rtt = step(&mut client, &mut server).unwrap().unwrap();
     assert!(!server.is_handshaking());
-    assert_eq!(server_1rtt, client_1rtt);
-    assert_ne!(server_hs, server_1rtt);
+    assert!(equal_secrets(&server_1rtt, &client_1rtt));
+    assert!(!equal_secrets(&server_hs, &server_1rtt));
     assert!(step(&mut client, &mut server).unwrap().is_none());
     assert!(step(&mut server, &mut client).unwrap().is_none());
 
-    // key update
-    let initial = quic::Secrets {
-        // Constant dummy values for reproducibility
-        client: vec![
-            0xb8, 0x76, 0x77, 0x08, 0xf8, 0x77, 0x23, 0x58, 0xa6, 0xea, 0x9f, 0xc4, 0x3e, 0x4a,
-            0xdd, 0x2c, 0x96, 0x1b, 0x3f, 0x52, 0x87, 0xa6, 0xd1, 0x46, 0x7e, 0xe0, 0xae, 0xab,
-            0x33, 0x72, 0x4d, 0xbf,
-        ],
-        server: vec![
-            0x42, 0xdc, 0x97, 0x21, 0x40, 0xe0, 0xf2, 0xe3, 0x98, 0x45, 0xb7, 0x67, 0x61, 0x34,
-            0x39, 0xdc, 0x67, 0x58, 0xca, 0x43, 0x25, 0x9b, 0x87, 0x85, 0x06, 0x82, 0x4e, 0xb1,
-            0xe4, 0x38, 0xd8, 0x55,
-        ],
-    };
-    let updated = client.update_secrets(&initial.client, &initial.server);
```

Comment:
@Ralith Please check out the code coverage report. You removed this use of `update_secrets`, which seemed to be the only test code using it, so its coverage went to zero. I think there may be other code coverage issues.

Reply:
I see in the commit where you mention this that "This is redundant to TLS key schedule tests." Does that mean that the `update_secrets` function is also redundant and can/should be removed? It seems like we either need to test it or remove it.

Reply:
Ah, forgot that part was still isolated from the TLS impl. IIRC it's in principle the same thing as `KeySchedule::derive_next`, and ideally there wouldn't be two distinct functions. However, some refactoring to QuicExt will be necessary for that to be feasible. In the mean time, I brought the test back in https://github.com/briansmith/rustls/pull/2.